### PR TITLE
#1360 Add tests for DiscoveryMojo

### LINKED ISSUE #1360: DiscoverMojo tests are absent
I think we have to add at least some tests for `org.eolang.maven.DiscoverMojo`. It will help with unerstanding for newcomers how `DiscoverMojo` actually works. Moreover, with tests we will able to change `DiscoverMojo` design securely.

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Add tests for `DiscoveryMojo`. Closes #1360

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp pls, take a look

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        final Map<String, String> first = json.removeFirst();
+        MatcherAssert.assertThat(
+            String.valueOf(dependencies),
+            Matchers.equalTo(first.get("discovered"))
+        );
+        MatcherAssert.assertThat(dependencies, Matchers.equalTo(json.size()));
+    }
+
+    /*
+     * @todo #1360:30min Make logger inside DiscoveryMojo non-static.
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:78]: @volodya-lombrozo How is logger related to DiscoverMojo functionality?

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:78]: I can't mock it in order to write test with true check

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        "org/eolang/maven/withwarning.eo, 1"
+    })
+    void executeDiscoveryPhaseForEoCorrectPrograms(
+        final String program,
+        final int dependencies,
+        final @TempDir Path tmp
+    ) throws IOException {
+        this.saveProgram(tmp, new ResourceOf(program));
+        this.discover(tmp);
+        final Deque<Map<String, String>> json = this.discoveredJsonEntries(tmp);
```
Comment (yegor256) [inline eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:68]: @volodya-lombrozo check this out: https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java (hunk 1 above)
Comment (mximp) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:78]: @volodya-lombrozo why would you need to mock a logger for tests? Can't it be set up appropriately for tests if needed?

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java (hunk 2 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:69]: @yegor256 `json` variable is using several times (not only once), I can't inline it. Anyway, I've fixed some other variables. Please, take a look.

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:79]: @mximp What do you mean? Do we have any other way to ask the logger about his last calls? Can you provide an example please? it's not so clear for me what is "set uo appropriately".

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+     */
+    private static final String EO_FOREIGN = "eo-foreign.json";
+
+    @ParameterizedTest
+    @CsvSource({
+        "org/eolang/maven/mess.eo, 7",
+        "org/eolang/maven/sum.eo, 0",
+        "org/eolang/maven/withwarning.eo, 1"
+    })
+    void executeDiscoveryPhaseForEoCorrectPrograms(
```
Comment (yegor256) [inline eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:62]: @volodya-lombrozo just a small hint: it's better to name test methods as if you finish a sentence that start with "The object under test...". In your case it would be: "The object under test _execute_ discovery phase for eo correct programs". It's better to say "executes"

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java (hunk 1 above)
Comment (yegor256) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:79]: @volodya-lombrozo I also feel that logging functionality is something orthogonal to object discovery functionality. We should not test logging in this test class, I believe. Also, it would be impossible to "inject logger as a field" since we use `com.jcabi.log.Logger` utility class everywhere, which redirects all calls to Maven logging facility.

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:79]: @yegor256 @mximp I don't wanna test the logger behavior, I wanna test `DiscoveryMojo` corner cases and how it behaves in different situations. It can be useful to see in tests a full picture. I often use tests as a class documentation and in tests I can see how the class actually works. If we have only successful paths in tests, I don't see the another important part and I have to open the class and read the full class code in order to understand his behaviour in corner cases. Moreover it increases test coverage.

Ok, for now I don't see any possibility to test it. I will remove the pazzle and disabled tests

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java]: Done

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java (hunk 3 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java]: @yegor256 Done. `executeDiscoveryPhaseForEoCorrectPrograms` -> `executesDiscoveryPhaseForCorrectEoPrograms`

**Comment by yegor256** [comment]
@rultor merge
